src/react/components/common/errorHandler/errorHandler.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import { IAppError, ErrorCode, AppError } from "../../../../models/applicationState";
+import { strings } from "../../../../common/strings";
+import Alert from "../alert/alert";
+
+/**
+ * Component properties for ErrorHandler component
+ */
+export interface IErrorHandlerProps extends React.Props<ErrorHandler> {
+    error: IAppError;
+    onError: (error: IAppError) => void;
+    onClearError: () => void;
+}
+
+/**
+ * Component for catching and handling global application errors
+ */
+export class ErrorHandler extends React.Component<IErrorHandlerProps> {
+    constructor(props, context) {
+        super(props, context);
+
+        this.onWindowError = this.onWindowError.bind(this);
+        this.onUnhandedRejection = this.onUnhandedRejection.bind(this);
+    }
+
+    public componentDidMount() {
```

Comment:
Consistently handles all uncaught exceptions whether they are sync/async

Reply:
how does this work?

Reply:
Due to event bubbling all errors/exceptions bubble up the call stack.  If nothing handles the error the finally land at the browser "window".  In this case we are attaching event handlers for all errors either from "error" (sync) and "unhandledrejection" (async).  We are then preventing the default behavior of logging to the console and calling our redux action to set the current error.